spec/package.spec.json

```diff
@@ -0,0 +1,278 @@
+          "$ref": "#/definitions/CompilerInformation"
+        },
+        "link_dependencies": {
+          "title": "The values for the link references found within this contract instances runtime bytecode",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/LinkValue"
+          }
+        }
+      }
+    },
+    "LinkValue": {
+      "title": "A value for an individual link reference in a contract's bytecode",
+      "type": "object",
+      "required": [
+        "offset",
+        "value"
+      ],
+      "properties": {
+        "offset": {
+          "type": "integer",
+          "minimum": 0
+        },
+        "value": {
+          "title": "The value for the link reference",
+          "type": "string",
+          "anyOf": [
+            {"$ref": "#/definitions/Address"},
+            {"$ref": "#/definitions/ContractInstanceName"},
+            {"$ref": "#/definitions/PackageContractInstanceName"}
+          ]
+        }
+      }
+    },
+    "ContractInstanceName": {
+      "title": "The name of the deployed contract instance",
+      "type": "string",
+      "pattern": "^[a-zA-Z][a-zA-Z0-9_]*$"
+    },
+    "PackageContractInstanceName": {
+      "title": "The path to a deployed contract instance somewhere down the dependency tree",
+      "type": "string",
+      "pattern": "^([a-z][-a-z0-9]{0,213}\\:)+[a-zA-Z][a-zA-Z0-9_]*$"
+    },
+    "CompilerInformation": {
+      "title": "Information about the software that was used to compile a contract type or instance",
+      "type": "object",
+      "required": [
+        "type",
+        "version"
+      ],
+      "properties": {
+        "type": {
+          "title": "The name of the compiler",
+          "enum": [
+            "solc",
+            "solcjs"
+          ]
+        },
+        "version": {
+          "title": "The version string for the compiler",
+          "type": "string"
+        },
+        "settings": {
+          "title": "The settings used for compilation",
+          "anyOf": [
+            {"$ref": "solc_Settings"}
+          ]
+        }
+      }
+    },
+    "solc_Settings": {
+      "title": "Settings for use with the solc or solcjs compiler",
+      "type": "object",
+      "properties": {
+        "optimize": {
+          "type": "boolean"
+        },
+        "optimize_runs": {
+          "type": "integer",
+          "minimum": 1
+        }
+      }
+    },
+    "Address": {
+      "title": "An Ethereum address",
+      "type": "string",
+      "pattern": "^0x[0-9a-fA-F]{40}$"
+    },
+    "TransactionHash": {
+      "title": "An Ethereum transaction hash",
+      "type": "string",
+      "pattern": "^0x[0-9a-zA-Z]{64}$"
+    },
+    "BlockHash": {
+      "title": "An Ethereum block hash",
+      "type": "string",
+      "pattern": "^0x[0-9a-zA-Z]{64}$"
+    },
+    "IPFS-URI": {
+      "title": "An IPFS URI",
+      "type": "string",
+      "format": "uri",
+      "pattern": "^ipfs:/?/?.*$"
+    }
+  }
+}
```

Comment:
Should this also get rid of the old json spec?

Reply:
Yeah probably, I'll get rid of it now. I can't think of any reason we'd want to keep it around